src/main/java/org/elasticsearch/index/translog/fs/FsTranslog.java

```diff
@@ -190,135 +238,162 @@ public long ramBytesUsed() {
 
     @Override
     public long translogSizeInBytes() {
-        FsTranslogFile current1 = this.current;
-        if (current1 == null) {
-            return 0;
+        long size = 0;
+        try (ReleasableLock _ = readLock.acquire()) {
+            size += current.sizeInBytes();
+            for (FsChannelReader translog : uncomittedTranslogs) {
+                size += translog.sizeInBytes();
+            }
         }
-        return current1.translogSizeInBytes();
+        return size;
     }
 
-    @Override
-    public int clearUnreferenced() {
-        rwl.writeLock().lock();
-        int deleted = 0;
-        try {
+    /** asserts that all files were closed, if not throws an {@link AssertionError} with details regarding the open files */
+    public static void assertAllClosed() {
+        if (ChannelReference.openedFiles == null) {
+            return;
+        }
+        RuntimeException exampleAllocator = null;
+        ArrayList<String> files = new ArrayList<>();
+        for (Map.Entry<String, Map<FsChannelReader, RuntimeException>> file : ChannelReference.openedFiles.entrySet()) {
+            files.add(file.getKey());
+            for (RuntimeException allocator : file.getValue().values()) {
+                if (exampleAllocator == null) {
+                    exampleAllocator = new RuntimeException(file.getKey() + " is still open", allocator);
```

Comment:
cool, I didn't know about this one. However it doesn't do quite as much as it will use the RuntimeException as a suppression and it's an unneeded wrapper (the allocator is descriptive enough).